src/test/java/cloud/fogbow/ras/core/cloudconnector/LocalCloudConnectorTest.java

```diff
@@ -860,83 +902,401 @@ public void testGetUserComputeQuota() throws FogbowException {
+    // resource type than the COMPUTE, NETWORK, VOLUME, ATTACHMENT, and PUBLIC_IP,
+    // an UnexpectedException will be thrown.
+    @Test(expected = UnexpectedException.class) // verify
+    public void testGetInstanceForAnUnsupportedResourceType() throws FogbowException {
+        // set up
+        Order<?> order = Mockito.mock(PublicIpOrder.class);
+        Mockito.when(order.getType()).thenReturn(ResourceType.GENERIC_RESOURCE);
+
+        // exercise
+        super.localCloudConnector.getInstance(order);
+    }
+    
+    // test case: When calling the deleteInstance method, the public IP plug-in must
+    // be called to exclude the public IP instance in the cloud from the instance ID
+    // contained in the order passed by parameter.
+    @Test
+    public void testDeletePublicIpInstance() throws FogbowException {
+        // set up
+        Order<?> order = Mockito.mock(PublicIpOrder.class);
+        Mockito.when(order.getType()).thenReturn(ResourceType.PUBLIC_IP);
+        Mockito.when(order.getInstanceId()).thenReturn(BaseUnitTests.FAKE_INSTANCE_ID);
+
+        // exercise
+        super.localCloudConnector.deleteInstance(order);
+
+        // verify
+        Mockito.verify(this.publicIpPlugin, Mockito.times(1)).deleteInstance(Mockito.any(),
+                Mockito.any(CloudUser.class));
+    }
+    
+    // test case: When invoking the getAllImages method, the plugin should throw an
+    // InstanceNotFoundException if it does not find the instance.
+    @Test
+    public void testDeleteInstanceThrowsInstanceNotFoundException() throws FogbowException {
+        // set up
+        Order<?> order = Mockito.mock(PublicIpOrder.class);
+        Mockito.when(order.getType()).thenReturn(ResourceType.PUBLIC_IP);
+        Mockito.when(order.getInstanceId()).thenReturn(BaseUnitTests.FAKE_INSTANCE_ID);
+
+        Mockito.doThrow(InstanceNotFoundException.class).when(this.publicIpPlugin).deleteInstance(Mockito.any(),
+                Mockito.any(CloudUser.class));
 
         // exercise
-        localCloudConnector.genericRequest(Mockito.anyString(), systemUser);
+        super.localCloudConnector.deleteInstance(order);
 
         // verify
-        Mockito.verify(systemToCloudMapperPlugin,
-                Mockito.times(1)).map(Mockito.any(SystemUser.class));
-        Mockito.verify(genericRequestPlugin, Mockito.times(1)).
-                redirectGenericRequest(Mockito.any(String.class), Mockito.eq(tokenMock));
+        Mockito.verify(this.publicIpPlugin, Mockito.times(1)).deleteInstance(Mockito.any(),
+                Mockito.any(CloudUser.class));
+    }
+    
+    // test case: When invoking the deleteInstance method of an order with a
+    // different resource type than the COMPUTE, NETWORK, VOLUME, ATTACHMENT,
+    // and PUBLIC_IP, an UnexpectedException will be thrown.
+    @Test(expected = UnexpectedException.class) // verify
+    public void testDeleteInstanceForAnUnsupportedResourceType() throws FogbowException {
```

Comment:
Maybe you should wrap the call in a try{}catch{} block and verify if the thrown message is as expected.

Reply:
Okay, I'll fix this.